client/src/components/DataFiles/DataFilesModals/DataFilesUploadModal.js

```diff
@@ -49,8 +49,14 @@ const DataFilesUploadModal = ({ className, layout }) => {
         }
       });
   };
-  const disabled =
+  const dropZoneDisabled =
     Object.values(status).filter(s => s === 'UPLOADING').length > 0;
+  const uploadButtonDisabled =
+    dropZoneDisabled ||
+    (!dropZoneDisabled &&
+      uploadedFiles.length ===
```

Comment:
No change needed, but fyi I think `!dropZoneDisabled` is redundant here
```suggestion
    (uploadedFiles.length ===
```

Reply:
Button and Drop File Zone are disabled when a file is uploading = dropZoneDisabled. That's why that validation is needed.

Reply:
But the line that precedes this one is `dropZoneDisabled || `, which implies that anything that comes after is `!dropZoneDisabled`